leader-election-impl/src/test/java/com/palantir/paxos/SqlitePaxosStateLogMigrationStateTest.java

```diff
@@ -44,43 +44,62 @@ public void setup() {
     }
 
     @Test
-    public void initialStateIsNotMigrated() {
-        assertThat(migrationState.hasAlreadyMigrated()).isFalse();
+    public void initialStateTest() {
+        assertThat(migrationState.hasMigratedFromInitialState()).isFalse();
+        assertThat(migrationState.isInValidationState()).isFalse();
+        assertThat(migrationState.isInMigratedState()).isFalse();
+    }
+
+    @Test
+    public void canSetStateToValidation() {
+        migrationState.migrateToValidationState();
+        assertThat(migrationState.hasMigratedFromInitialState()).isTrue();
+        assertThat(migrationState.isInValidationState()).isTrue();
+        assertThat(migrationState.isInMigratedState()).isFalse();
     }
 
     @Test
     public void canSetStateToMigrated() {
-        migrationState.finishMigration();
-        assertThat(migrationState.hasAlreadyMigrated()).isTrue();
+        migrationState.migrateToMigratedState();
+        assertThat(migrationState.hasMigratedFromInitialState()).isTrue();
+        assertThat(migrationState.isInValidationState()).isFalse();
+        assertThat(migrationState.isInMigratedState()).isTrue();
     }
 
     @Test
-    public void canSetStateToMigratedMultipleTimes() {
-        migrationState.finishMigration();
-        migrationState.finishMigration();
-        migrationState.finishMigration();
-        assertThat(migrationState.hasAlreadyMigrated()).isTrue();
+    public void canChangeStates() {
+        migrationState.migrateToValidationState();
+        migrationState.migrateToMigratedState();
+        migrationState.migrateToMigratedState();
+        assertThat(migrationState.hasMigratedFromInitialState()).isTrue();
+        assertThat(migrationState.isInValidationState()).isFalse();
+        assertThat(migrationState.isInMigratedState()).isTrue();
+
+        migrationState.migrateToValidationState();
+        assertThat(migrationState.hasMigratedFromInitialState()).isTrue();
+        assertThat(migrationState.isInValidationState()).isTrue();
+        assertThat(migrationState.isInMigratedState()).isFalse();
```

Comment:
Last 4 lines would be different if we're not allowing migrated -> validated state.